### ISSUE TITLE
The new `init()` to substitute `enclose()` with different args in v2.0

### ISSUE THREAD

**Issue by koresar** [ericelliott/stampit#62]
`enclose()` function name confuses people. Make the new `init()` function an alias to `enclose()` (or vice versa). It should receive an object with options.

* **instance**
* **stamp**
* **args**

Thanks to ES6 destructuring, using them is almost as convenient as using formal params:

```js
stamp.init(({ instance, stamp, args }) => {
  console.log(instance, stamp, args);
});
```

This way we leave the created hook open for extension like `.enclose()` should have been.

This will be useful in many scenarios.

##### `clone()` method for object instances
```js
var cloneable = stampit.init(function (options) {
  var stamp = options.stamp;

  // add .clone() method to prototype (which is 'fixed.methods')
  stamp.fixed.methods.clone = function () { return stamp(this); };

  // or add .clone() method to each object istance
  options.instance.clone = function () { return stamp(this); }
  // same as above
  this.clone = function () { return stamp(this); }
});

var myStamp = stampit(null, { data: 42 }).compose(cloneable);
var instance = myStamp.create().clone().clone().clone().clone().clone();
assert(instance.data === 42);
```

##### `getStamp()` method for object instances
```js
var stampProvidable = stampit.init(function (options) {
  var stamp = options.stamp;

  // add .getStamp() method to prototype
  stamp.fixed.methods.getStamp = function () { return stamp; };

  // or add .getStamp() method to each object istance
  options.instance.getStamp = function () { return stamp; }
  // same as above
  this.getStamp = function () { return stamp; }
});

var myStamp = stampit().compose(stampProvidable);
var stampFromInstance = myStamp.create().getStamp();
assert(myStamp === stampFromInstance);
```

**Comment by ericelliott** [comment]
:+1:

**Comment by unstoppablecarl** [comment]
Will `args` be an array? or an `arguments` object? can you provide an example of its use?

**Comment by ericelliott** [comment]
I think it should be a real array, for convenience. The `arguments` object sucks.

**Comment by unstoppablecarl** [comment]
Personally I would prefer stampit factories to accept a single object hash param, and you could always assume that to be the case. I think an object of sufficient complexity to use stampit would likely accept enough arguments to necessitate a single settings object (this is how I tend to do it at least). This would also remove the need to have to handle the `arguments` object that can hurt performance and cause headaches.

**Comment by ericelliott** [comment]
Stamps can take arbitrary arguments so that you can use them with legacy constructors.

The problem with that is constructor arguments can collide with each other. );

This issue has already been debated ad nauseam. After exactly ten billion requests, I relented and washed my hands of it.

It's going to be hard to change now. People are relying on the current behavior. I know that we're looking at a 2.0 release and breaking changes are OK, but we still don't want to make the migration path too steep.. =)

**Comment by koresar** [comment]
We are loosing `convertConstructor` parameters support because of this feature. This means that we are going back to the old days when stampit supported only parameterless constructors. We can always reimplement that feature in the future releases on stampit 2.0.

@ericelliott JFYI ^^

**Comment by koresar** [comment]
Or, actually we are not. Disregard the message above. All good now. :)

**Comment by ericelliott** [comment]
:+1:

**Comment by koresar** [comment]
Implemented for v2.0 with #72.
